### DISCUSSION TITLE
command line options for dosemu?

### DISCUSSION THREAD

**Discussion by xucaen** [dosemu2/dosemu2#1652] [Q&A]
Is there a list of Dosemu2 command line options? Specifically I'm looking for a way to pass a socket into Dosemu2 and I'm looking for how to specify the autoexec.bat (because so far placing autoexec.bat in c:\ doesn't do anything). But a list of options would be good to have to help figure these out.

**Comment by stsp** [comment]
man dosemu
man dosemu.bin
dosemu --help (somewhat outdated)

I have no idea what is "pass socket to
dosemu2".
Use userhook.bat as autoexec.bat.

**Reply by xucaen** [reply]
I assume you know what a socket is (part the C standard library). I wrote a tcp server that accepts connections on a port, and creates a socket. Then I can pass that socket onto other programs so they can communicate over tcp (like telnet). So I was assuming that, because dosemu2 has a config setting for com1="vmodem" and comes with dosemu FOSSIL emulator, that there must be a way for my server to pass - something - into dosemu so that my dos program will have an active connection over com1. I have done this with dosbox, however dosbox will not work with my dos programs. I'm not sure how else to explain it.  I checked man dosemu.bin and I don't see a way to pass in a socket or file handle, so I'm guessing it can't be done.

Do you know of any way to run a BBS in dosemu and have it accept telnet connections? That might be a way for me to do what I want.

**Reply by stsp** [reply]
Plenty of ways.
dosemu2 has a built-in
TCP server over a hayes
modem protocol - although
the incoming connections
are not supported at a moment.

Or you can use `socat` with
your own server.

**Reply by xucaen** [reply]
I can't find "socat" in the man pages for either dosemu or dosemu.bin. What is it?

**Reply by stsp** [reply]
Since you seem to insist
on your own server instead
of dosemu2's vmodem,
see socat on creating a
link between pty and your
socket.

**Reply by stsp** [reply]
> I can't find "socat" in the man pages for either dosemu or dosemu.bin. What is it?

How about `man socat`?

**Reply by stsp** [reply]
Also as far as I can recall,
the ability to work with pipes
and sockets was added directly.
So what exactly is a problem
when you pass your socket
to $_com1 option?

**Reply by andrewbird** [reply]
Socat is a Linux general solution, it should be available via a package on your distro.

**Reply by xucaen** [reply]
Thanks for the info on socat, I never heard of it before so I'll research that this week. 

when you say "...when you pass your socket to $_com1 option" - that's exactly my question - is there an option to pass my socket into dosemu? How do I pass my socket into $_com1? I don't see the option in man dosemu.bin

**Reply by stsp** [reply]
Its not a command-line
option, its a config option.
Of course as you claim
that local configs do not
work for you, good luck
with socat instead.
Oh wait, its the same
thing: with socat you'll 
end up passing pty to
$_com1.

So better get your .dosemurc
working first.

**Reply by xucaen** [reply]
Ok so you're saying that 
a) I could write "$_com1=" some-random-filehandle to the config file and it will connect $_com1 to the socket, and
b) you guys have already implemented the code to allow $_com1 to point to a file handle

yes, the /etc/dosemu/dosemu.conf works perfectly, I have no problems editing that file at runtime in order to store the socket as $_com1, if you say it will work. Can you show me the documentation on it? I don't want to begin work on the parser just to find out you guys didn't actually implement the feature.

Look, you're documentation isn't that good. I'd volunteer to help write it, but I'll need you to give me the information I need first. can I have to source code for dosemu2? Maybe if I had that I wouldn't have to trouble you with questions.

**Reply by stsp** [reply]
Are you serious?
You say you come to github
and have no idea what is it
and where to get the code
of the hosted projects?

https://github.com/dosemu2/dosemu2/wiki/Building-From-Source

**Reply by stsp** [reply]
Also dosemu.conf is pretty well
self-documented in its comments.
So what prevents you from just
checking if that works or not?

Also I can see that you no
longer use mailer to reply, so
perhaps you can find out a
way of attaching the files
finally. :)

**Comment by andrewbird** [comment]
Perhaps this example of using socat with dosemu2 can help? https://github.com/dosemu2/dosemu2/issues/406#issuecomment-596145135
